### PR TITLE
Updating the User Model for username, email max_length change

### PR DESCRIPTION
**Pull request by jay-zenefits**
@usmanghani @glynn-zenefits 

Note: This is the django User model change.
I have not updated the email field on the existing migrations since it is already done here. https://github.com/zenefits/django/commit/7fd55c3481a004afb049e15ae3b8c93ce8bf0603

### CODE REVIEW

**Comment by jay-zenefits** [comment]
@zenefits/csg - FYI

File: django/contrib/auth/migrations/0001_initial.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ class Migration(migrations.Migration):
                 ('password', models.CharField(max_length=128, verbose_name='password')),
                 ('last_login', models.DateTimeField(default=timezone.now, verbose_name='last login')),
                 ('is_superuser', models.BooleanField(default=False, help_text='Designates that this user has all permissions without explicitly assigning them.', verbose_name='superuser status')),
-                ('username', models.CharField(help_text='Required. 30 characters or fewer. Letters, digits and @/./+/-/_ only.', unique=True, max_length=30, verbose_name='username', validators=[validators.RegexValidator('^[\\w.@+-]+$', 'Enter a valid username.', 'invalid')])),
+                ('username', models.CharField(help_text='Required. 254 characters or fewer. Letters, digits and @/./+/-/_ only.', unique=True, max_length=254, verbose_name='username', validators=[validators.RegexValidator('^[\\w.@+-]+$', 'Enter a valid username.', 'invalid')])),
```
Comment (usmanghani) [inline django/contrib/auth/migrations/0001_initial.py]: Why are you changing an existing migration?

File: django/contrib/auth/migrations/0004_alter_user_username_opts.py [hunk 2]
```diff
@@ -16,6 +16,6 @@ class Migration(migrations.Migration):
         migrations.AlterField(
             model_name='user',
             name='username',
-            field=models.CharField(error_messages={'unique': 'A user with that username already exists.'}, max_length=30, validators=[django.core.validators.RegexValidator('^[\\w.@+-]+$', 'Enter a valid username. This value may contain only letters, numbers and @/./+/-/_ characters.', 'invalid')], help_text='Required. 30 characters or fewer. Letters, digits and @/./+/-/_ only.', unique=True, verbose_name='username'),
+            field=models.CharField(error_messages={'unique': 'A user with that username already exists.'}, max_length=254, validators=[django.core.validators.RegexValidator('^[\\w.@+-]+$', 'Enter a valid username. This value may contain only letters, numbers and @/./+/-/_ characters.', 'invalid')], help_text='Required. 254 characters or fewer. Letters, digits and @/./+/-/_ only.', unique=True, verbose_name='username'),
```
Comment (usmanghani) [inline django/contrib/auth/migrations/0004_alter_user_username_opts.py]: Why are you changing an existing migration instead of make migrations?

**Comment by usmanghani** [comment]
Sorry never mind just realized its already 254 in production.
and initdb.

**Comment by usmanghani** [comment]
Just realized this is just bringing this in synchronization with the state of the world

**Comment by usmanghani** [comment]
@jessecollier  any issues with this?

**Comment by jessecollier** [comment]
This might conflict with future upgrades of Django, fyi. 

Also you updated migration files (which won't actually make any change)... since they are already applied

**Comment by jay-zenefits** [comment]
ya.. we dint want another migration coz we updated the db already.

And since its our version of django, if we get a conflict during our next upgrade, we can do the same fix there.

**Comment by jay-zenefits** [comment]
And this is the real PR https://github.com/zenefits/django/pull/22 which got merged to the right branch.

**Pull request merged**